braces/views/_other.py

```diff
@@ -124,3 +124,25 @@ def dispatch(self, request, *args, **kwargs):
 
         handler = getattr(self, self.all_handler, self.http_method_not_allowed)
         return handler(request, *args, **kwargs)
+
+
+class HeaderMixin(object):
+    """
+    Add arbitrary HTTP headers to a response by specifying them in the
+    ``headers`` attribute or by overriding the ``get_headers()`` method.
+    """
+    headers = {}
+
+    def get_headers(self, request):
+        return self.headers
+
+    def dispatch(self, request, *args, **kwargs):
+        """
+        Override this method to customize the way additional headers are
+        retrieved.  It is mandatory that the returned value supports the
+        ``.items()`` method.
+        """
+        response = super(HeaderMixin, self).dispatch(request, *args, **kwargs)
+        for key, value in self.get_headers(request).items():
+            response[key] = value
```

Comment:
`setdefault` is only available on Django 1.8 and later. I'll add a check to prevent existing headers from being overwritten.